### DISCUSSION TITLE
Cast

### DISCUSSION THREAD

**Discussion by Squad-G** [tpetry/laravel-query-expressions#8] [Ideas]
Hello

I was looking at ways to protect raw data / nicer to read and I found this package on Laravel-News (https://laravel-news.com/query-expressions-for-laravel).

This is a really good package and so far I love it.

There's one problem I ran into... I think it's missing the cast function.

Right Now, I got the following code:
`new Expression("sum(cast({$this->balanceWithoutTaxesField} as float)) as balance_without_taxes"),`

It would be nice if it could be refactored with something like this:
`new Alias(new Sum(new Cast($this->balanceWithoutTaxesField, 'float')), 'balance_without_taxes')`

The type could be an Enum but it would lock the package to 8.1

Thanks and keep up the good work

**Comment by tpetry** [comment]
I'll look at whether casting can be done in a platform-neutral way.

**Comment by Squad-G** [comment]
Thank you 🙏

**Comment by Plytas** [comment]
Any news on this?

**Reply by tpetry** [reply]
I didn't see a way to do this in a way that it works on all systems yet. Their types are all so different, with barely a common thing.

But I could start with a very limited one. To which types do you want to cast currently?

**Reply by Plytas** [reply]
We're using MySQL and we need to cast to int/float equivalents. We could create custom expressions if you could point us the right way.

**Reply by tpetry** [reply]
I think casting to numeric values is the most-safe way. You can look at the current implementations of the expressions on how to do it manually. Will play with casting to int/float later the day to see whether the reduced feature-set would work.

**Comment by tpetry** [comment]
I've implemented a first basic implementation for numeric types. You can now cast to integer (`int`, `bigint`) and floating-point (`float`, `double`) types with `v.1.3.0`.

/cc @Squad-G @Plytas
